miniconf/load_site_data.py

```diff
@@ -0,0 +1,204 @@
+    site_data: Dict[str, Any],
+    by_uid: Dict[str, Any],
+    qa_session_length_hr: int,
+) -> List[str]:
+    """Loads all site data at once.
+
+    Populates the `committee` and `by_uid` using files under `site_data_path`.
+
+    NOTE: site_data[filename][field]
+    """
+    extra_files = ["README.md"]
+    # Load all for your sitedata one time.
+    for f in glob.glob(site_data_path + "/*"):
+        extra_files.append(f)
+        name, typ = f.split("/")[-1].split(".")
+
+        if name == "acl2020_accepted_papers":
+            continue
+
+        if typ == "json":
+            site_data[name] = json.load(open(f))
+        elif typ in {"csv", "tsv"}:
+            site_data[name] = list(csv.DictReader(open(f)))
+        elif typ == "yml":
+            site_data[name] = yaml.load(open(f).read(), Loader=yaml.SafeLoader)
+
+    for typ in ["papers", "speakers"]:
+        by_uid[typ] = {}
+        for p in site_data[typ]:
+            by_uid[typ][p["UID"]] = p
+
+    display_time_format = "%H:%M"
+
+    # index.html
+    site_data["committee"] = build_committee(site_data["committee"]["committee"])
+
+    # schedule.html
+    site_data["schedule"] = build_plenary_sessions(site_data["speakers"])
+
+    # tutorials.html
+    tutorials = build_tutorials(site_data["tutorials"])
+    site_data["tutorials"] = tutorials
+    # tutorial_<uid>.html
+    by_uid["tutorials"] = {
+        tutorial.id: tutorial
+        for tutorial in tutorials
+    }
+
+    # papers.html
+    build_papers(site_data, by_uid, display_time_format, qa_session_length_hr)
+    # poster_<uid>.html
+    posters = build_posters(site_data["papers"])
+    # papers.json
+    site_data["posters"] = posters
+    by_uid["posters"] = {
+        poster.id: poster
+        for poster in posters
+    }
+
+    # workshops.html
+    workshops = build_workshops(site_data["workshops"])
+    site_data["workshops"] = workshops
+    # workshop_<uid>.html
+    by_uid["workshops"] = {
+        workshop.id: workshop
+        for workshop in workshops
+    }
+
+    # sponsors.html
+    build_sponsors(site_data, by_uid, display_time_format)
+
+    # about.html
+    site_data["faq"] = site_data["faq"]["FAQ"]
+    site_data["code_of_conduct"] = site_data["code_of_conduct"]["CodeOfConduct"]
+
+    print("Data Successfully Loaded")
+    return extra_files
+
+
+def build_committee(raw_committee: List[Dict[str, Any]]) -> List[CommitteeMember]:
+    return [jsons.load(item, cls=CommitteeMember) for item in raw_committee]
+
+
+def build_plenary_sessions(raw_keynotes: List[Dict[str, Any]]) -> Dict[str, Keynote]:
+    # TODO: define a better dataclass
+    return {
+        day: {
+            "speakers": [item for item in raw_keynotes if item["day"] == day]
+        }
+        for day in ["Monday", "Tuesday", "Wednesday"]
+    }
+
+
+def build_papers(site_data, by_uid, display_time_format: str, qa_session_length_hr: int) -> None:
+    for session_name, session_info in site_data["poster_schedule"].items():
```

Comment:
this method should be refactored in the next PR when we unify the main conference papers, demo papers, and SRW papers